metadata/images.go

```diff
@@ -189,11 +217,24 @@ func (s *imageStore) Delete(ctx context.Context, name string, opts ...images.Del
 		return err
 	}
 
-	return withImagesBucket(s.tx, namespace, func(bkt *bolt.Bucket) error {
-		err := bkt.DeleteBucket([]byte(name))
+	return update(ctx, s.db, func(tx *bolt.Tx) error {
+		bkt, err := createImagesBucket(tx, namespace)
```

Comment:
Since this is a delete, shouldn't this use `getImageBukcet()` and ignore any "ErrBucketNotFound" errors? (or maybe not ignore them, not sure if delete is supposed to be idempotent at this level).

I believe right now it would create the the bucket if it doesn't exist, which is a bit unexpected.

Reply:
I agree, that is not correct behavior. The transaction ends up getting rolled back anyway but it is unnecessary to create.